CI/dagger-pipeline.py

```diff
@@ -0,0 +1,125 @@
+"""Run notebook tests within a docker pipeline to test integration."""
+
+import logging
+import os
+import sys
+from pathlib import Path
+
+try:
+    import anyio
+    import dagger
+    from dagger import BuildArg
+except ImportError:
+    raise ImportError(
+        "Dagger is not installed. Please install it with `pip install dagger-io` first."
+    )
+
+try:
+    from dagger._engine._version import CLI_VERSION as __dagger_version__
+except ModuleNotFoundError:
+    __dagger_version__ = "unknown"
+
+
+BASE_IMAGE_TAG = os.getenv("BASE_IMAGE_TAG")
+PAVICS_HOST = os.getenv("PAVICS_HOST", "https://pavics.ouranos.ca")
+SANITIZE_FILE_URL = os.getenv(
+    "SANITIZE_FILE_URL",
+    "https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/raw/master/notebooks/output-sanitize.cfg",
+)
+
+
+async def main():
+    async with dagger.Connection(
+        dagger.Config(
+            log_output=sys.stderr, workdir=Path(__file__).parent.parent.as_posix()
+        )
+    ) as client:
+        if not BASE_IMAGE_TAG:
+            raise ValueError("BASE_IMAGE_TAG environment variable is not set.")
+
+        sources = (
+            await (
+                # pull container
+                client.container().from_(f"pavics/workflow-tests:{BASE_IMAGE_TAG}")
+                # copy files to container
+                .with_directory(
+                    "/code",
```

Comment:
This repo is checked out under `/code`, inside the image `pavics/workflow-tests:{BASE_IMAGE_TAG}`?